src/content/learn/understanding-your-ui-as-a-tree.md

```diff
@@ -0,0 +1,268 @@
+import Greeting from './Greeting';
+import Copyright from './Copyright';
+
+export default function App() {
+  return (
+    <>
+      <Title text="Generate a random greeting" />
+      <Greeting>
+        <Copyright year={2004} />
+      </Greeting>
+    </>
+  );
+}
+
+```
+
+```js GreetingMedia.js
+export function Title({fancy, text}) {
+  return <h1 className={fancy ? 'fancy' : ''}>{text}</h1>;
+}
+
+export function Image({src}) {
+  return <img src={src} />;
+}
+```
+
+```js Greeting.js
+import * as React from 'react';
+import greetings from './greetings';
+import {Title, Image} from './GreetingMedia';
+
+function randIndex(max) {
+  return Math.floor(Math.random() * max);
+}
+
+export default function Greeting({children}) {
+  const [index, setIndex] = React.useState(randIndex(greetings.length));
+  const greeting = greetings[index];
+  const nextGreeting = () => setIndex(randIndex(greetings.length));
+  return (
+    <>
+      {greeting.type === 'text' ? (
+        <Title fancy text={greeting.value} />
+      ) : (
+        <Image src={greeting.value} />
+      )}
+      <button onClick={nextGreeting}>🎲</button>
+      {children}
+    </>
+  );
+}
+
+```
+
+```js Copyright.js
+export default function Copyright({year}) {
+  return <p>©️ {year}</p>;
+}
+```
+
+```js greetings.js
+export default [
+    {type: 'text', value: 'Howdy!'},
+    {type: 'text', value: 'Hello!'},
+    {type: 'image', value: 'https://i.imgur.com/yXOvdOSs.jpg'},
+    {type: 'image', value: 'https://i.imgur.com/yXOvdOSs.jpg'},
+    {type: 'text', value: 'Howdy!'},
+];
+```
+
+```css
+.fancy {
+    font-style: italic;
+    color: blue;
+}
+```
+</Sandpack>
+
+<Diagram name="conditional_render_tree" height={300} width={522} alt="TODO.">
+
+With conditional rendering, across different renders, the render tree may render different components.
+
+</Diagram>
+
+In this example, depending on what `greeting.type` is, we may render `<Title>` or `<Image>`. The render tree may be different for each render pass.
+
+Although render trees may differ across render pases, these trees are generally helpful for identifying what the top-level components are in a React app. Top-level components affect the rendering performance of all the components beneath them and often contain the most complexity.
+
+## The Module Dependency Tree {/*the-module-dependency-tree*/}
+
+Another relationship in a React app that can be modeled with a tree are an app's module dependencies. As we [break up our components](/learn/importing-and-exporting-components#exporting-and-importing-a-component) and logic into separate files, we create [JS modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) where we may export components, functions or constants. 
+
+Each node in a module dependency tree is a module and each branch represents an `import` statement in that module.
+
+If we take the previous Greeting app, we can build a module dependency tree, or dependency tree for short. 
```

Comment:
>  we can build a module dependency tree, or dependency tree for short

Is a "dependency tree" distinct from a "module dependency tree" in other languages? this might be confusing to someone coming from another background if "dependency tree" has meaning outside of what you're describing here.

Reply:
Hmm, I personally don't think it'd be a source of confusion even for other languages as we're setting the context that we're using it as a shorthand for module dependency tree here. if you feel strongly, happy to change!